cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+            chunksizes=hex_str_to_ndarray(chunksizes),
+            timestamps=hex_str_to_ndarray(timestamps),
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, BitstringAccumulator):
+            return False  # coverage: ignore
+
+        if (
+            self.max_setting != other.max_setting
+            or self.simul_settings != other.simul_settings
+            or self.circuit_params != other.circuit_params
+            or self.qubit_to_index != other.qubit_to_index
+        ):
+            return False  # coverage: ignore
+
+        if not np.array_equal(self.bitstrings, other.bitstrings):
+            return False  # coverage: ignore
+
+        if not np.array_equal(self.chunksizes, other.chunksizes):
+            return False  # coverage: ignore
+
+        if not np.array_equal(self.timestamps, other.timestamps):
+            return False  # coverage: ignore
+
+        return True
+
+    def summary_string(self, setting: InitObsSetting, number_fmt='.3f'):
+        return (
+            f'{setting}: {self.mean(setting):{number_fmt}} +- {self.stderr(setting):{number_fmt}}'
+        )
+
+    def __repr__(self):
+        return (
+            f'cirq.work.BitstringAccumulator('
+            f'meas_spec={self.meas_spec!r}, '
+            f'simul_settings={self.simul_settings!r}, '
+            f'qubit_to_index={self.qubit_to_index!r}, '
+            f'bitstrings={proper_repr(self.bitstrings)}, '
+            f'chunksizes={proper_repr(self.chunksizes)}, '
+            f'timestamps={proper_repr(self.timestamps)})'
+        )
+
+    def __str__(self):
+        s = f'Accumulator {self.max_setting}; {self.n_repetitions} repetitions\n'
+        s += '\n'.join('  ' + self.summary_string(setting) for setting in self._simul_settings)
+        return s
+
+    def covariance(self, *, atol=1e-8) -> np.ndarray:
+        """Compute the covariance matrix for the estimators of all settings.
+
+        Like `variance`, this is the covariance of the sampling distribution
+        of the sample mean. Practically, it is the 'normal' covariance
+        divided by the number of observations (bitstrings).
+
+        Args:
+            atol: The absolute tolerance for asserting coefficients are real.
+        """
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
+            raise ValueError("No measurements")
+
+        all_obs_vals = np.array(
+            [
+                _obs_vals_from_measurements(
+                    bitstrings=self.bitstrings,
+                    qubit_to_index=self._qubit_to_index,
+                    observable=setting.observable,
+                    atol=atol,
+                )
+                for setting in self._simul_settings
+            ]
+        )
+
+        # https://github.com/numpy/numpy/issues/11502
+        if all_obs_vals.shape[0] == 1:
+            # coverage: ignore
+            cov = np.array([[np.var(all_obs_vals[0], ddof=1)]])
```

Comment:
Is there no way to trigger this in a test?

Reply:
done